File: PersianCalendar/src/main/java/com/byagowi/persiancalendar/di/dependencies/MainActivityDependency.java
package com.byagowi.persiancalendar.di.dependencies;

import com.byagowi.persiancalendar.di.scopes.PerActivity;
import com.byagowi.persiancalendar.ui.MainActivity;

import javax.inject.Inject;

@PerActivity
public final class MainActivityDependency {
    @Inject
    MainActivity activity;

    @Inject
    public MainActivityDependency() {
    }

    public MainActivity getMainActivity() {
        return activity;
    }
}

File: PersianCalendar/src/main/java/com/byagowi/persiancalendar/di/dependencies/MainActivityDependency.kt
package com.byagowi.persiancalendar.di.dependencies

import com.byagowi.persiancalendar.di.scopes.PerActivity
import com.byagowi.persiancalendar.ui.MainActivity

import javax.inject.Inject

@PerActivity
class MainActivityDependency @Inject
constructor() {
    @Inject
    lateinit var mainActivity: MainActivity internal set
}
